internal/run/source.go

```diff
@@ -167,7 +177,7 @@ func (w *Workflow) RunSource(ctx context.Context, parentStep *workflowTracking.W
 
 		inSchemas := []string{}
 		for _, input := range source.Inputs {
-			resolvedPath, err := schema.ResolveDocument(ctx, input, nil, mergeStep)
+			resolvedPath, err := schemas.ResolveDocument(ctx, input, nil, mergeStep)
```

Comment:
So I need the result of ResolveDocument for the given input (which may be many different things based on the logic here) to be passed to the method in sdk-gen-config

At the moment, that call happens long before all of this logic to determine the correct input. This leads me to think that a lot of the logic here should belong in that repo - it's quite hard to follow at the moment. Is there something exceptional that makes it so that the natural home should be here?

Reply:
Im having trouble following you. Should we discuss tomorrow morning?